tests/TranslatorTest.php

```diff
@@ -11,10 +11,41 @@ class TranslatorTest extends TestCase
 {
     private const INVALID = 'INVALID';
 
+    /**
+     * @dataProvider provideGetCleanBranch
+     */
+    public function testGetCleanBranch(string $branch, string $expected)
+    {
+        $actual = $this->invokeMethod('getCleanBranch', $branch);
+        $this->assertSame($expected, $actual);
+    }
+
+    public function provideGetCleanBranch()
+    {
+        return [
+            [
+                '5.0',
+                '5.0'
+            ],
+            [
+                'invalid',
+                'invalid'
+            ],
+            [
+                '(HEAD detached at 1.2.3)',
+                '1.2'
+            ],
+            [
+                '(HEAD detached at 4.5.6-beta1)',
+                '4.5'
+            ]
```

Comment:
```suggestion
            ],
            [
                '(HEAD detached at 123456)',
                '(HEAD detached at 123456)'
            ],
```
Covers the case where a specific commit hash is checked out

Reply:
Updated